dev/PushNotifications/PushNotificationsLongRunningTask/platform.cpp

```diff
@@ -46,14 +47,61 @@ void NotificationsLongRunningPlatformImpl::WaitForWinMainEvent()
     m_shutdownTimerManager->Wait();
 }
 
-// Example of one function. We will add more as we need them.
-STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterFullTrustApplication(_In_ PCWSTR /*processName*/, _In_ GUID /*remoteId*/, _Out_ GUID* /*appId*/) noexcept
+void NotificationsLongRunningPlatformImpl::AddToRegistry(const std::wstring& processName, std::wstring appId)
+{
+    std::wstring subkey = L"Software\\Microsoft\\Windows\\CurrentVersion\\PushNotifications\\LongRunningProcess";
```

Comment:
`PCWSTR`

You make a string here (ctor, malloc, memcpy) so you can just .c_str() and then destroy it (dtor, free). Expensive way to have a string literal.

C++ is cool, but C++ isn't always the best answer ;-)

Reply:
Using ApplicationData instead of Registry. Removed.